### PR TITLE
Pretty-print durations in log messages

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds simple pretty-printing to duration in log timeout messages. If the duration is >= 1 second, it's printed in seconds with a fractional part. If the duration is less than 1 second, it is printed in milliseconds. This simple formatting may not be sufficient as a formatting rule for all cases, but should be sufficient for printing our relatively small timeouts.

Log messages now look something like this:
```
ERROR 2018-04-04T20:05:49Z: conduit_proxy: turning operation timed out after 100 ms into 500
```

Previously, they looked like this:
```
ERROR 2018-04-04T20:07:26Z: conduit_proxy: turning operation timed out after Duration { secs: 0, nanos: 100000000 } into 500
```

I made this change partially because I wanted to make the panics from the `eventually!` macro added in #669 more readable.

### CODE REVIEW

File: proxy/src/timeout.rs [hunk 1]
```diff
@@ -212,3 +219,43 @@ where
            .finish()
     }
 }
+
+//===== impl HumanDuration =====
+
+impl fmt::Display for HumanDuration {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        // let precision = fmt.precision().unwrap_or(3);
```
Comment (olix0r) [inline proxy/src/timeout.rs]: ?

Comment (hawkw) [reply proxy/src/timeout.rs]: oops 😓

File: proxy/src/timeout.rs [hunk 2]
```diff
@@ -212,3 +219,42 @@ where
...
 }
+
+//===== impl HumanDuration =====
+
+impl fmt::Display for HumanDuration {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        let secs = self.as_secs();
+        let subsec_ms = self.subsec_nanos() as f64 / 1_000_000f64;
+        if secs == 0 {
+            write!(fmt, "{} ms", subsec_ms)
```
Comment (olix0r) [inline proxy/src/timeout.rs]: tioli: i'd prefer to omit the space between the value and unit so log lines end up as e.g.

> `ERROR 2018-04-04T20:05:49Z: conduit_proxy: turning operation timed out after 100ms into 500`

File: proxy/src/timeout.rs [hunk 3]
```diff
@@ -35,6 +34,13 @@ pub struct TimeoutFuture<F> {
     timeout: ReactorTimeout,
 }
 
+/// Pretty-print durations as fractional seconds.
```
Comment (olix0r) [inline proxy/src/timeout.rs]: nit: "Pretty-prints durations"---descriptive, not imperative.

File: proxy/src/timeout.rs [hunk 4]
```diff
@@ -153,8 +159,9 @@ where
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
             TimeoutError::Timeout(ref duration) =>
-                // TODO: format the duration nicer.
-                write!(f, "operation timed out after {:?}", duration),
+                write!(f, "operation timed out after {}",
+                    HumanDuration(*duration)
+                ),
```
Comment (olix0r) [inline proxy/src/timeout.rs]: also, i'd rather let rustfmt sort all of this out, but i prefer wrapping multiline match blocks with`{}` (even when there's a single statement).

Also, I think we should standardize at 100char line breaks, which I think this means we could collapse this onto a single line (especially if we rename `duration` to `d`, which seems fine in this context).

Comment (hawkw) [reply proxy/src/timeout.rs]: Okay, I prefer 80-character lines but can accept 100! :)

**Pull request merged**